src/Shipping/ShippingZone.php

```diff
@@ -78,6 +77,89 @@ public function get_shipping_methods_for_country( string $country_code ): array
 		return ! empty( $this->methods_countries[ $country_code ] ) ? array_values( $this->methods_countries[ $country_code ] ) : [];
 	}
 
+	/**
+	 * Gets the shipping rates for the given country.
+	 *
+	 * @param string $country_code
+	 *
+	 * @return array[] A multidimensional array of shipping rates for the given country. {
+	 *     Array of shipping method arguments.
+	 *
+	 *     @type string $country  The shipping method country.
+	 *     @type string $method   The shipping method ID.
+	 *     @type string $currency The currency which the shipping rate is in. Defaults to the store currency.
+	 *     @type float  $rate     The cost of the shipping method. Only if the method is flat-rate.
+	 *     @type array  $options  Array of options for the shipping method (varies based on the method type). {
+	 *         Array of options for the shipping method.
+	 *
+	 *         @type array[] $shipping_class_rates An array containing the shipping class names and their rates. Only if the method is flat-rate.
+	 *         @type float   $free_shipping_threshold The minimum order amount required to use the shipping method. Only if the method is free shipping.
+	 *
+	 *     }
+	 * }
+	 */
+	public function get_shipping_rates_for_country( string $country_code ): array {
+		$methods = $this->get_shipping_methods_for_country( $country_code );
+
+		// If the only shipping method is the free shipping method, return it as the only rate.
+		if ( 1 === count( $methods ) && self::METHOD_FREE === $methods[0]['id'] ) {
+			$rate = [
+				'country'  => $country_code,
+				'method'   => $methods[0]['id'],
+				'currency' => $methods[0]['currency'],
+				'rate'     => 0,
+				'options'  => [],
+			];
+			if ( isset( $methods[0]['options']['min_amount'] ) ) {
+				$rate['options']['free_shipping_threshold'] = $methods[0]['options']['min_amount'];
+			}
+
+			return [ $rate ];
+		}
```

Comment:
Thanks for adding this though I think the problem is not fixed yet, imagine this scenario where you have a country with:

- Free Shipping (enable)
- Flat rate (disabled)

It will not sync the free shipping because count(methods) = 2. 

What do you think if we filter all the disabled methods before we continue with the execution and we could have a method to `parse_rates_contry` so we could re-use the code and it will be easier to test. 

```
$methods = $this->get_shipping_methods_for_country( $country_code );
$methods_enabled = filter only enable methods 

if ( 1 === count( $methods_enabled ) && self::METHOD_FREE === $methods_enabled[0]['id'] ) {
  return [$this->parse_rates_country( $methods_enabled[0] )]

}


```

Reply:
Oh you're right. Didn't consider that case. I've added some commits including unit tests to prevent that issue.

Reply:
The structure that we expect here is a bit different than what is returned by `parse_method`. This function returns the rates as expected by our API so the data structure is changed to be similar to how we store the rates in DB.